polaris-icons/package.json

```diff
@@ -63,5 +63,8 @@
     "js-yaml": "^4.0.0",
     "rehype-parse": "^7.0.1",
     "unified": "^9.2.1"
+  },
+  "peerDependencies": {
+    "react": "*"
```

Comment:
Any insights on why this was introduced?

Reply:
`@shopify/polaris-icons` produces a build that depends on React as a peer dependency but was not declaring it. Take for example this file that it exports: https://www.unpkg.com/browse/@shopify/polaris-icons@8.9.0/dist/icons/AdjustIcon.svg.mjs

```js
import React from 'react';

var SvgAdjustIcon = function SvgAdjustIcon(props) {
  return /*#__PURE__*/React.createElement("svg", Object.assign({
    viewBox: "0 0 20 20"
  }, props), /*#__PURE__*/React.createElement("path", {
    fillRule: "evenodd",
    d: "M9.095 6.25a3.001 3.001 0 0 1 5.81 0h1.345a.75.75 0 0 1 0 1.5h-1.345a3.001 3.001 0 0 1-5.81 0h-5.345a.75.75 0 0 1 0-1.5h5.345Zm1.405.75a1.5 1.5 0 1 1 3 0 1.5 1.5 0 0 1-3 0Z"
  }), /*#__PURE__*/React.createElement("path", {
    fillRule: "evenodd",
    d: "M8 16a3.001 3.001 0 0 0 2.905-2.25h5.345a.75.75 0 0 0 0-1.5h-5.345a3.001 3.001 0 0 0-5.81 0h-1.345a.75.75 0 0 0 0 1.5h1.345a3.001 3.001 0 0 0 2.905 2.25Zm1.5-3a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0Z"
  }));
};
SvgAdjustIcon.displayName = "AdjustIcon";

export { SvgAdjustIcon as S };
```

It imports `react` but did not specify it as a peer dependency, which meant that under pnpm it would not have access to it for producing its build and resulted in errors when linking it.

This can be shipped separately from this PR if you'd like me to split it out, its an issue with the current state of the package, pnpm just made it more apparent